cadence/contracts/FlowEVMPassthrough.cdc

```diff
@@ -0,0 +1,153 @@
+            self.borrowCOA().deposit(from: <-flowVault)
+        }
+
+        access(FungibleToken.Withdrawable) fun withdraw(amount: UFix64): @{FungibleToken.Vault} {
+            return <-self.borrowCOA().withdraw(
+                balance: EVM.Balance(flow: amount)
+            )
+        }
+
+        /// Deposits Flow to defined EVM Address using a call to passthrough Solidity contract
+        access(all) fun evmTransfer(from: @{FungibleToken.Vault}, to: EVM.EVMAddress, gasLimit: UInt64) {
+            pre {
+                self.isSupportedVaultType(type: from.getType()): "Passthrough only supports FlowToken.Vaults"
+            }
+            let amount = from.getBalance()
+            // TODO: Replace with UInt256(EVM.Balance(flow: from.balance).toAttoFlow())
+            let amountInAttoFlow = FlowEVMPassthrough.ufix64ToUInt256(value: from.getBalance(), decimals: 8)
+            // TODO: Replace with EVM.encodeABIWithSignature("transferFlow(address)", [to])
+            let calldata = FlowEVMPassthrough.encodeABIWithSignature(
+                    "transferFlow(address)", 
+                    [to]
+                )
+            let coa = self.borrowCOA()
+            let flowVault <- from as! @FlowToken.Vault
+            coa.deposit(from: <-flowVault)
+            coa.call(
+                to: FlowEVMPassthrough.passthroughEVMAddress,
+                data: calldata,
+                gasLimit: gasLimit,
+                value: EVM.Balance(flow: amount)
+            )
+
+            emit Transfered(from: self.borrowCOA().address(), to: to, amount: amount)
+        }
+
+        access(self) fun borrowCOA(): &EVM.BridgedAccount {
+            return self.coaCapability.borrow() ?? panic("COA Capability has been revoked and is no longer valid")
+        }
+    }
+
+    /* --- Internal Helpers --- */
+    // TODO: Remove helpers once EVM.Balance.toAttoFlow() is implemented
+
+    /// Raises the base to the power of the exponent
+    access(self) view fun pow(base: UInt256, exponent: UInt8): UInt256 {
+        if exponent == 0 {
+            return 1
+        }
+        var r = base
+        var exp: UInt8 = 1
+        while exp < exponent {
+            r = r * base
+            exp = exp + 1
+        }
+        return r
+    }
+
+    /// Converts a UFix64 to a UInt256
+    access(self) view fun ufix64ToUInt256(value: UFix64, decimals: UInt8): UInt256 {
+        let integerPart: UInt64 = UInt64(value)
+        var r = UInt256(integerPart)
+        var multiplier: UInt256 = self.pow(base:10, exponent: decimals)
+        return r * multiplier
+    }
+
+    access(self) fun encodeABIWithSignature(
+        _ signature: String,
+        _ values: [AnyStruct]
+    ): [UInt8] {
+        let methodID = HashAlgorithm.KECCAK_256.hash(
+            signature.utf8
+        ).slice(from: 0, upTo: 4)
+        let arguments = EVM.encodeABI(values)
+
+        return methodID.concat(arguments)
+    }
```

Comment:
These can be removed once we get an EVM CLI pre-release that includes the updated EVM contract